server/src/main/java/org/opensearch/search/aggregations/bucket/terms/InternalSignificantTerms.java

```diff
@@ -291,7 +296,12 @@ protected B reduceBucket(List<B> buckets, ReduceContext context) {
         List<InternalAggregations> aggregationsList = new ArrayList<>(buckets.size());
         for (B bucket : buckets) {
             subsetDf += bucket.subsetDf;
-            supersetDf += bucket.supersetDf;
+            // supersetDf is a shard level count, so we don't sum it for segment slices
```

Comment:
I think it would actually produce `bucketCount * actual_supersetDf` since there is not necessarily 1 bucket per slice, but I agree that this could use some more explanation here